### ISSUE TITLE
[BUG] Issues finding the glfw header file, Compile Error and as small Makefile example issues

### ISSUE THREAD

**Issue by 42ajakob** [codam-coding-college/MLX42#113]
For this errors I used a freshly installed MLX42, the Makefile and so_long.c example in Basics.md.

MLX42 has issues finding the headers file of my glfw which I installed with 42homebrew and it's in the .brew directory. Although cmake finds it while executing Findglfw3.cmake file when the files are compiling I get this error message:

In file included from MLX42/src/font/mlx_font.c:14:
./MLX42/include/MLX42/MLX42_Int.h:23:11: fatal error: 'GLFW/glfw3.h' file not found
        # include <GLFW/glfw3.h>
        
I basically fixed this issue by putting "-I ~/.brew/Cellar/glfw/3.3.8/include/" into my HEADERS variable in the Makefile.

Also would be nice if we can add "$ENV{HOME}/homebrew/include/" and "$ENV{HOME}/homebrew/lib/" to the Findglfw3.cmake file for some people that install brew the "Untar anywhere (unsupported)" way which tells you to install it in ~/homebrew directory.

After the first error fixed with modifing the HEADERS variable in Makefile when I try to compile I get the error message:

MLX42/src/mlx_images.c:90:17: error: comparison of integers of different signs: 'int32_t' (aka 'int') and 'size_t' (aka 'unsigned long') [-Werror,-Wsign-compare]
        if (img->count >= ctx->instances_capacity)

No idea how to fix it and it seems to me like a library issue.

And the last little thing for the Makefile example would be in the "clean" command  to add a -r flag to:
       @rm -f $(LIBMLX)/build

Since it's a directory with stuff in it that you can't remove without this flag. cmake also creates the directory if it doesn't exist with the -B flag so no need to change more. 

I tried it on the 42 Mac at school

**Comment by W2Wizard** [comment]
Thanks for reporting this, currently a bit pre-occupied with my internship.

TL;DR
- [ ] Add `$ENV{HOME}/homebrew/include/`
- [ ] Add `$ENV{HOME}/homebrew/lib/`
- [ ] Silly sign compare in `mlx_images.c:90:17`
- [ ] Add the `-r` flag in the clean command for the Makefile in docs

If you. would be up to a making a PR that addresses these fixes I could simply merge them for you. If not, give me a bit.
